src/Native/JsArray.js

```diff
@@ -0,0 +1,84 @@
+// import Maybe
+
+var _elm_lang$core$Native_JsArray = function() {
+/* A thin, but still immutable, wrapper over native Javascript arrays. */
+
+// An empty array
+var empty = {
+    ctor: 'JsArray',
+    _0: []
+};
+
+function singleton(val) {
+    return {
+        ctor: 'JsArray',
+        _0: [val]
+    };
+}
+
+function length(arr) {
+    return arr._0.length;
+}
+
+function get(idx, arr) {
+    var val = arr._0[idx];
+    if (val) {
+        return _elm_lang$core$Maybe$Just(val);
+    }
+
+    return _elm_lang$core$Maybe$Nothing;
+}
+
+function set(idx, val, arr) {
+    var copy = arr._0.slice();
+    copy[idx] = val;
```

Comment:
This will put elements freely in the array. E.g -. 
![screen shot 2016-07-25 at 00 36 07](https://cloud.githubusercontent.com/assets/1139198/17086969/d8b41d26-51ff-11e6-8da7-62b7da415caf.png)